overlord/servicestate/quota_control.go

```diff
@@ -83,6 +84,59 @@ func quotaGroupsAvailable(st *state.State) error {
 	return nil
 }
 
+func isQuotaAvailable(st *state.State, feature features.SnapdFeature, settingName string) error {
+	tr := config.NewTransaction(st)
+	status, err := features.Flag(tr, feature)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if !status {
+		return fmt.Errorf("experimental feature disabled - test it by setting 'experimental.%s' to true", settingName)
+	}
+	return nil
+}
+
+func verifyQuotaRequirements(st *state.State, resourceLimits quota.Resources) error {
+	// validate quotas in general are available
+	if err := quotaGroupsAvailable(st); err != nil {
+		return err
+	}
+
+	// Upon initialization verification for systemd version 230 has already been done,
+	// but for some of these quota types we need even higher version.
+	// see: EnsureQuotaUsability
+
+	// MemoryLimit requires systemd 211, so it's covered by the initial check
+	// CPUQuota requires systemd 213, so no further checks need to be done
+	// TasksMax requires systemd 228, so no further checks need to be done
+
+	// AllowedCPUs requires systemd 243, so we need to verify the version here
+	if resourceLimits.CPUSet != nil {
+		if err := systemd.EnsureAtLeast(243); err != nil {
+			return fmt.Errorf("cannot use the cpu-set quota with incompatible systemd: %v", err)
+		}
+	}
+
+	// Journal quotas require systemd 245, so we need to verify the version here as well
+	if resourceLimits.Journal != nil {
+		if err := systemd.EnsureAtLeast(245); err != nil {
+			return fmt.Errorf("cannot use journal quota with incompatible systemd: %v", err)
+		}
+
+		// We also need to check against experimental features for journal quota
+		if journErr := isQuotaAvailable(st, features.JournalQuota, "journal-quota"); journErr != nil {
+			// For backwards compatability with uses of journal quota before this added feature
+			// we also allow the enabled quota-groups feature switch to be set
+			if err := isQuotaAvailable(st, features.QuotaGroups, "quota-groups"); err != nil {
```

Comment:
do we have a test for this? can we have it, or can we have it only later when we drop the other check?